anesthetic/samples.py

```diff
@@ -470,22 +457,63 @@ def ns_output(self, nsamples=200):
             Samples from the P(logZ, D, d) distribution
 
         """
-        dlogX = self._dlogX(nsamples)
+        dlogX = self.dlogX(nsamples)
+        samples = MCMCSamples(index=dlogX.columns)
+        samples['logZ'] = self.logZ(dlogX)
+
+        logw = dlogX.add(self.logL, axis=0)
+        logw -= samples.logZ
+        S = (dlogX*0).add(self.logL, axis=0) - samples.logZ
+
+        samples['D'] = numpy.exp(logsumexp(logw, b=S, axis=0))
+        samples['d'] = numpy.exp(logsumexp(logw, b=(S-samples.D)**2, axis=0))*2
+
+        samples.tex = {'logZ': r'$\log\mathcal{Z}$',
+                       'D': r'$\mathcal{D}$',
+                       'd': r'$d$'}
+        samples.label = self.label
+        return samples
+
+    def logZ(self, nsamples=None):
```

Comment:
Do the methods `self.logZ()`, `self.D()` and `self.d()` now conflict with the corresponding attributes `self.logZ` etc. created by `ns_output()`? Do we accept that or would it be better to rename the methods, e.g. `self.get_logZ()` or `self.calc_logZ()`?

Reply:
Fortunately they don't conflict. The object returned from `ns_output` is an `MCMCSamples` object, not a `NestedSamples` one, so there shouldn't be a conflict.